src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -0,0 +1,680 @@
+    {}
+
+    CHIP_ERROR UpdateKey(StorageKeyName & key) override
+    {
+        VerifyOrReturnError(kUndefinedFabricIndex != fabric_index, CHIP_ERROR_INVALID_FABRIC_INDEX);
+        key = DefaultStorageKeyAllocator::FabricSceneKey(fabric_index, index);
+        return CHIP_NO_ERROR;
+    }
+
+    void Clear() override { mStorageData.Clear(); }
+
+    CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override
+    {
+        CharSpan NameSpan(mStorageData.mName, mStorageData.mNameLength);
+        TLV::TLVType container;
+        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, container));
+
+        // Scene ID
+        ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneEndpointID), static_cast<uint16_t>(mStorageId.mEndpointId)));
+        ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneGroupID), static_cast<uint16_t>(mStorageId.mGroupId)));
+        ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneID), static_cast<uint8_t>(mStorageId.mSceneId)));
+
+        // Scene Data
+        // A length of 0 means the name wasn't used so it won't get stored
+        if (!NameSpan.empty())
+        {
+            ReturnErrorOnFailure(writer.PutString(TLV::ContextTag(kTagSceneName), NameSpan));
+        }
+
+        ReturnErrorOnFailure(
+            writer.Put(TLV::ContextTag(kTagSceneDTransitionTime), static_cast<uint16_t>(mStorageData.mSceneTransitionTimeSeconds)));
+        ReturnErrorOnFailure(
+            writer.Put(TLV::ContextTag(kTagSceneDTransitionTime100), static_cast<uint8_t>(mStorageData.mTransitionTime100ms)));
+        ReturnErrorOnFailure(mStorageData.mExtensionFieldSets.Serialize(writer));
+
+        return writer.EndContainer(container);
+    }
+
+    CHIP_ERROR Deserialize(TLV::TLVReader & reader) override
+    {
+        CharSpan NameSpan(mStorageData.mName);
+
+        ReturnErrorOnFailure(reader.Next(TLV::AnonymousTag()));
+        VerifyOrReturnError(TLV::kTLVType_Structure == reader.GetType(), CHIP_ERROR_INTERNAL);
+
+        TLV::TLVType container;
+        ReturnErrorOnFailure(reader.EnterContainer(container));
+
+        // Scene ID
+        ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneEndpointID)));
+        ReturnErrorOnFailure(reader.Get(mStorageId.mEndpointId));
+        ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneGroupID)));
+        ReturnErrorOnFailure(reader.Get(mStorageId.mGroupId));
+        ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneID)));
+        ReturnErrorOnFailure(reader.Get(mStorageId.mSceneId));
+
+        // Scene Data
+        ReturnErrorOnFailure(reader.Next());
+        TLV::Tag currTag = reader.GetTag();
+        VerifyOrReturnError(TLV::ContextTag(kTagSceneName) == currTag || TLV::ContextTag(kTagSceneDTransitionTime) == currTag,
+                            CHIP_ERROR_WRONG_TLV_TYPE);
+
+        // If there was no error, a name is expected from the storage, if there was an unexpectec TLV element,
```

Comment:
This comment seems weirdly truncated... and a name is not always expected from storage.  How about:
```suggestion
        // A name may or may not have been stored.  Check whether it was.
```